crates/oxc_language_server/src/main.rs

```diff
@@ -492,21 +530,46 @@ impl Backend {
         if config.exists() {
             config_path = Some(config);
         }
+
+        let mut oxlintrc = None;
+        let mut config_store = None;
+        let mut linter = self.server_linter.write().await;
         if let Some(config_path) = config_path {
-            let mut linter = self.server_linter.write().await;
-            let config = Oxlintrc::from_file(&config_path)
-                .expect("should have initialized linter with new options");
-            let config_store = ConfigStoreBuilder::from_oxlintrc(true, config.clone())
-                .expect("failed to build config")
-                .build()
-                .expect("failed to build config");
+            oxlintrc = Some(
+                Oxlintrc::from_file(&config_path)
+                    .expect("should have initialized linter with new options"),
+            );
+            config_store = Some(
+                ConfigStoreBuilder::from_oxlintrc(true, oxlintrc.clone().unwrap())
+                    .expect("failed to build config")
+                    .build()
+                    .expect("failed to build config"),
+            );
+        }
+
+        if self.options.lock().await.use_nested_configs {
```

Comment:
If using nested configs, use `Linter::new_with_nested_configs` and pass in the nested config files. Otherwise, should be no behavior change. However, I did do some refactoring here to make this easier for me to read through.